plugins/guests/coreos/guest.rb

```diff
@@ -4,7 +4,7 @@ module VagrantPlugins
   module GuestCoreOS
     class Guest < Vagrant.plugin("2", :guest)
       def detect?(machine)
-        machine.communicate.test("cat /etc/os-release | grep ID=coreos")
+        machine.communicate.test("(cat /etc/os-release | grep ID=coreos) || (cat /etc/os-release | grep ID_LIKE=coreos)")
```

Comment:
According to the [man page](https://www.freedesktop.org/software/systemd/man/os-release.html#ID_LIKE=) ID_LIKE is a list, so it can be `ID_LIKE=rhel fedora`. So in this case, `ID_LIKE=whatever coreos` would fail the check.

We should do something like `grep -E 'ID_LIKE=.*coreos.*'`.

Reply:
done in updated commit.